test/Altinn.App.Api.Tests/Mocks/InstanceClientMockSi.cs

```diff
@@ -243,8 +243,9 @@ public async Task<Instance> UpdateSubstatus(int instanceOwnerPartyId, Guid insta
 
             if (substatus == null || string.IsNullOrEmpty(substatus.Label))
             {
-                throw await PlatformHttpException.CreateAsync(
-                    new HttpResponseMessage { StatusCode = System.Net.HttpStatusCode.BadRequest });
+                using var response = 
+                    new HttpResponseMessage { StatusCode = System.Net.HttpStatusCode.BadRequest };
+                throw await PlatformHttpException.CreateAsync(response);
```

Comment:
If I interpret you right you are pointing out what I worry about too about this, which is that the exception will have a reference to a disposed response, whereas it didn't before. Before this PR I think

* Catchers of this exception could potentially read the response, since we didn't dispose it (I'm assuming we're either leaking the underlying resource now or that it has a finalizer that disposes upon GC)

After this PR
* Attempts to do anything with the content of the response in a catch block will throw an exception, while e.g. `StatusCode` still works fine (not guarded by `bool _disposed`)

If you're talking about the order of operations here, which is also potentially confusing, `CreateAsync` is fully completed before the response is disposed (so `CreateAsync` will have no problem reading the content)

Reply:
This discussion also plays into `HttpCompletionOption`. Default behavior (and the only behavior we use AFAICT) of `HttpClient` is to buffer responses into a `MemoryStream`, in which case the `Dispose` implementation doesn't seem to do anything apart from marking the stream as disposed and no longer readable.

In the `HttpCompletionOption.ResponseHeadersRead` case however, there is more work being done on dispose and I think this is where it is particularly important to clean up.

We could do this cleanup as part of v9 instead, and make a breaking change on `PlatformHttpException` so that it doesn't expose the whole response (only the information which is safely referenced even after dispose)

Reply:
I think you are right. I was just confused why you skipped adding the `using` statements in other places this issue happened, but did here. Thinking more I don't think it's an issue.